app/controllers/metadata_controller.rb

```diff
@@ -42,11 +42,19 @@ def validate_csv_for_new_samples
       )
     end
 
-    issues = validate_metadata_csv_for_samples(samples, metadata, true)
+    issues, new_host_genomes = validate_metadata_csv_for_new_samples(
```

Comment:
For the `validate_csv_for_new_samples` method, I would add the comment:


If validation passes and the user provides host genome names that we currently don't support, new custom host genomes will be created and assigned to that user.
This is done at this step because the sample upload flow requires host genome ids to be passed to the create endpoint.